packages/jest-cli/src/prepareStackTrace.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+// filter for noisy stack trace lines
+const FILTER_NOISE =
+  /(\/|\\)jest(-.*?)?(\/|\\)(vendor|build|node_modules|packages)(\/|\\)/;
+
+module.exports = (context: {Error: Class<Error>}) => {
+  const Error = context.Error;
+  Error.prepareStackTrace = (error: Error, stacktrace: CallSite[]): mixed => {
+    const filteredStacktrace = stacktrace.filter(
+      (callsite: CallSite, index: number) => {
+        return index === 0 || !FILTER_NOISE.test(callsite.getFileName() || '');
```

Comment:
```
  (callsite, index) => index === 0 || !FILTER_NOISE.test(callsite.getFileName() || '')
```

this should work, otherwise you can break at the `=>`. There is likely no need to annotate the arguments with types, flow will be able to infer them :)